src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -196,6 +196,8 @@ public void setDefinition(FlowDefinition definition) {
         } else {
             quietPeriod = null;
         }
+        getTriggersJobProperty().stopTriggers();
+        getTriggersJobProperty().startTriggers(Items.currentlyUpdatingByXml());
```

Comment:
Not really the right place to do this. `setOwner` seems more appropriate, but this was dropped in 09adb56241cbcac3baf2625f063af8da7ba87bee; perhaps @abayer recalls why.